tsl/src/chunk.c

```diff
@@ -334,6 +336,62 @@ chunk_create_replica_table(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+static void
+chunk_api_call_chunk_drop_replica(Chunk *chunk, const char *node_name, Oid serverid)
+{
+	const char *drop_cmd;
+	List *data_nodes;
+
+	/*
+	 * Drop chunk on the data node using a regular "DROP TABLE".
+	 * Note that CASCADE is not required as it takes care of dropping compressed
+	 * chunk (if any).
+	 *
+	 * If there are any other non-TimescaleDB objects attached to this table due
+	 * to some manual user activity then they should be dropped by the user
+	 * before invoking this function.
+	 */
+
+	drop_cmd = psprintf("DROP TABLE %s.%s",
+						quote_identifier(chunk->fd.schema_name.data),
+						quote_identifier(chunk->fd.table_name.data));
+	data_nodes = list_make1((char *) node_name);
+	ts_dist_cmd_run_on_data_nodes(drop_cmd, data_nodes, true);
+
+	/*
+	 * This chunk might have this data node as primary, change that association
+	 * if so. Then delete the chunk_id and node_name association.
+	 */
+	chunk_update_foreign_server_if_needed(chunk->fd.id, serverid);
+	ts_chunk_data_node_delete_by_chunk_id_and_node_name(chunk->fd.id, node_name);
+}
+
+void
+chunk_api_call_chunk_drop_replica_wrapper(ChunkMoveData *ccd, Oid serverid, bool transactional)
+{
+	if (transactional)
+		StartTransactionCommand();
+
+	chunk_api_call_chunk_drop_replica(ccd->chunk, ccd->src_node, serverid);
+
+	if (transactional)
+	{
+		NameData application_name;
+
+		snprintf(application_name.data,
+				 sizeof(application_name.data),
+				 "chunk_dropped_srcdn:%s",
+				 application_name.data);
+		pgstat_report_appname(application_name.data);
```

Comment:
Doesn't it make more sense to update the appname within `chunk_move_activity_update`? It seems code to update the appname is duplicated before every call to `chunk_move_activity_update`

Reply:
+1. I would also note, if this string is set as a process backend name (show in `ps/htop` for example) we should set something more meaningful to the user, like prefixing it with `move_chunk(): ...` or something like that, if makes sense.

I am not totally sure what it means to update application name here, for instance what if client is also using application name?

Reply:
The string that we see in ps/htop is fixed by Postgres. We cannot update it ourselves. See `QueryCompletion` in PG sources. I think we should avoid setting application name if the user has set it explicitly

Reply:
We can play around with `cluster_name` GUC instead of `application_name`. But `application_name` goes into logs as well, so might help in debugging and forensics after the fact.